build/build.go

```diff
@@ -1523,3 +1540,47 @@ func ReadSourcePolicy() (*spb.Policy, error) {
 
 	return &pol, nil
 }
+
+// backendAttribute is a utility to retrieve the backend attribute from a resolvedNode.
+func backendAttribute(dp *resolvedNode) attribute.KeyValue {
+	driverName := dp.Node().Driver.Factory().Name()
+	return attribute.String("backend", driverName)
+}
+
+// buildIDAttribute is a utility to retrieve the build id attribute from the solve options.
+// This value should be consistent between builds.
+func buildIDAttribute(so *client.SolveOpt) attribute.KeyValue {
+	vcs := so.FrontendAttrs["vcs:source"]
+	target := so.FrontendAttrs["target"]
+	context := so.FrontendAttrs["context"]
+	filename := so.FrontendAttrs["filename"]
+
+	buildID := ""
+	if vcs != "" || target != "" || context != "" || filename != "" {
+		h := xxh3.New()
+		for _, s := range []string{vcs, target, context, filename} {
+			h.WriteString(s)
+			h.Write([]byte{0})
+		}
+		buildID = hex.EncodeToString(h.Sum(nil))
```

Comment:
I think we need to add something unique to node into this mix. Eg. see `tryNodeIdentifier`. Otherwise, this metric can leak data with a simple plaintext attack.

Reply:
I think we need to be able to correlate the same parameters between different machines. I believe the node identifier is unique between machines. Would there be a different salt parameter we could use for this purpose that would still allow us to match build identities across machines?